### PR TITLE
Auth: Login and Signup

### PR DESCRIPTION
**Pull request by tushartiwari7**
- Added Login and Signup Pages.
- Also added services for the same.

### CODE REVIEW

**Review by Mohitkumar33 — APPROVED** [review]
clean and well structured code

File: src/components/Layout/Editor/Editor.jsx [hunk 1]
```diff
@@ -44,6 +55,11 @@ const Editor = () => {
         contentEditable
         className="editor__textarea input p-lg fs-m"
         placeholder="Start Typing..."
+        onInput={(e) =>
+          debounced(
+            Array.from(e.target.childNodes).map((node) => node.textContent)
```
Comment (dikshadhote) [inline src/components/Layout/Editor/Editor.jsx:60]: This was new for me

File: src/components/Layout/Editor/Editor.jsx [hunk 2]
```diff
@@ -26,6 +26,17 @@ const Editor = () => {
...
     },
   ];
+
+  const debounce = (func, delay) => {
+    let inDebounce;
+    return (val) => {
+      clearTimeout(inDebounce);
+      inDebounce = setTimeout(() => func(val), delay);
+    };
+  };
```
Comment (dikshadhote) [inline src/components/Layout/Editor/Editor.jsx:36]: like the fact that ,you made reusable debounce function

**Review by dikshadhote — APPROVED** [review]
Overall, well structured code .

**Pull request merged**